### PR TITLE
Test main

### PR DESCRIPTION
**Pull request by gustavste**
- sanity test of valkyrie using test config

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -33,20 +52,25 @@ func main() {
...
-		_, _ = fmt.Fprintf(os.Stdout, "Version: %s\n", appVersion)
-		return
+		_, _ = fmt.Fprintf(out, "Version: %s\n", appVersion)
+		return 0
 	}
 
 	cfg, err := configs.Read(configFilePath)
 	if err != nil {
-		log.Fatal().Err(err).Msg("Failed to read config")
+		log.Err(err).Msg("Failed to read config")
```
Comment (gnirb) [inline main.go:61]: Will logging an error automatically make it use error logging level (`log.Error()...`). Noted that you just removed `.Fatal()` without replacing it with another level

File: provider/internal/test/suite.go [hunk 2]
```diff
@@ -69,33 +70,41 @@ func (s *IntegrationTestSuite) SetupSuite() {
...
-				ProviderAddress: fmt.Sprintf(baseAddr, providerPort),
-				OperatorAddress: fmt.Sprintf(baseAddr, operatorPort),
-			},
-		}
-
-		s.valkyrie = server.NewValkyrie(context.TODO(), &valkyrieConfig)
+		valkyrieConfig := s.buildConfig(pamURL, providerPort, operatorPort, dataStore)
 
+		if v, err := server.NewValkyrie(context.TODO(), &valkyrieConfig); err != nil {
+			log.Error().Err(err)
```
Comment (gnirb) [inline provider/internal/test/suite.go:76]: I might be wrong here, but I think it needs a message (or call to `Send()`) to actually log the error

File: server/server.go [hunk 3]
```diff
@@ -80,35 +64,38 @@ func NewValkyrie(ctx context.Context, cfg *configs.ValkyrieConfig) *Valkyrie {
...
 
 	// Provider routes.
 	if err = routes.ProviderRoutes(v.provider, cfg, pamClient, httpClient); err != nil {
-		log.Fatal().Err(err).Msg("Unable to setup the intended provider routes")
+		log.Err(err).Msg("Unable to setup the intended provider routes")
+		return nil, err
 	}
 	if err = routes.OperatorRoutes(v.operator, cfg, httpClient); err != nil {
-		log.Fatal().Err(err).Msg("Unable to setup the intended operator routes")
+		log.Err(err).Msg("Unable to setup the intended operator routes")
```
Comment (gnirb) [inline server/server.go:77]: Similar to previous comment, is it fine to omit the logging level?

File: main_test.go [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+			"Operator server listening on 'localhost:",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.Name, func(tt *testing.T) {
+			tt.Setenv("VALK_PROFILES", "testlog")
+			p, _ := testutils.GetFreePort()
+			o, _ := testutils.GetFreePort()
+			tt.Setenv("PROVIDER_PORT", fmt.Sprintf("%d", p))
+			tt.Setenv("OPERATOR_PORT", fmt.Sprintf("%d", o))
```
Comment (gnirb) [inline main_test.go:53]: Do these override the configured port from config file?

Comment (gnirb) [reply main_test.go:53]: Ah strike that, now I see the config file values have the env vars inlined

File: configs/testdata/valkyrie_config.test.main.yml [hunk 5]
```diff
@@ -0,0 +1,71 @@
+logging:
```
Comment (gnirb) [inline configs/testdata/valkyrie_config.test.main.yml:1]: Is it possible to use the already existing example for config? Just concerned about keeping two almost identical config files up to date

Comment (gustavste) [reply configs/testdata/valkyrie_config.test.main.yml:1]: It was the problem with the ports. The test fails with the default ports

File: provider/internal/test/suite.go (hunk 2 above)
Comment (gustavste) [reply provider/internal/test/suite.go:76]: You're probably right. But also realised the error is already logged inside `server.NewValkyrie` so i'll put something else here

File: server/server.go (hunk 3 above)
Comment (gustavste) [reply server/server.go:77]: yes

File: main.go [hunk 6]
```diff
@@ -33,20 +52,25 @@ func main() {
...
+		return 1
```
[the remark this answers is not in the document]
Comment (gnirb) [reply main.go:62]: Alright, guess we use int too then 😄

File: configs/testdata/valkyrie_config.test.main.yml (hunk 5 above)
Comment (gnirb) [reply configs/testdata/valkyrie_config.test.main.yml:1]: Ah right, well guess its a separate issue to tackle if we keep changing these files a lot

Comment (gustavste) [reply configs/testdata/valkyrie_config.test.main.yml:1]: Maybe i can to the whole field a env parameter, and if it's missing the parsing will use the default value

**Review by gnirb — APPROVED** [review]

File: configs/testdata/valkyrie_config.test.main.yml (hunk 5 above)
Comment (gnirb) [reply configs/testdata/valkyrie_config.test.main.yml:1]: Sure, worth a shot

**Review by gnirb — APPROVED** [review]

**Pull request merged**